src/features/pathway/components/BuildingLifecycleCard.jsx

```
@@ -0,0 +1,257 @@
+      }
+    }
+    events.sort((a, b) => a.year - b.year);
+
+    // Count constructions for rebuild labels
+    let constructCount = 0;
+    for (const e of events) {
+      if (e.type === 'construct') {
+        constructCount++;
+        e.constructIndex = constructCount;
+      }
+      if (fixedStartYear == null && e.year < minYear) minYear = e.year;
```

ESLint

[error] Reassigning a variable after render has completed can cause inconsistent behavior on subsequent renders. Consider using state instead

(react-compiler/react-compiler)